### ISSUE TITLE
Incomplete inference of implicit arguments in the `exists` tactic

### ISSUE THREAD

**Issue by amahboubi** [coq/coq#13456]
#### Description of the problem
This script works in Coq 8.11.*, but the first line fails in 8.12.1.
```coq
Lemma minbug (n : nat) (P : nat -> Prop) (pn : P n) : exists (m : nat) (p : P m), True.
Proof.
  exists _, pn.
(* Error: Cannot infer this placeholder of type "nat"... *)
(* exists I. *)
Admitted.
```
#### Coq Version

8.12.1

**Comment by herbelin** [comment]
Hi @amahboubi, this is probably related to #12365 which should then be refined.

**Comment by herbelin** [comment]
Fixed in #13468. Thanks for reporting!

**Comment by amahboubi** [comment]
Thanks for the swift fix @herbelin !
